x-pack/plugins/reporting/server/lib/tasks/monitor_reports.ts

```diff
@@ -70,43 +64,58 @@ export class MonitorReportsTask implements ReportingTask {
     });
   }
 
+  /*
+   * This task finds report jobs that can get lost from the queue and must be recovered.
```

Comment:
How can a job get lost from the queue?

Reply:
There is a file-level comment that kind of explained it, but it wasn't doing a great job at it. 

4bb793d28e1 consolidates the comments and hopefully answers your question.